stock_picking_auto_create_package/models/stock_picking.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2023 Jacques-Etienne Baudoux (BCIM) <je@bcim.be>
+# Copyright 2023 Raumschmiede Gmbh
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, models
+from odoo.exceptions import UserError
+from odoo.tools.float_utils import float_round
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _auto_create_delivery_package(self) -> None:
+        self.ensure_one()
+        if self.picking_type_id.automatic_package_creation_mode == "single":
+            self._auto_create_delivery_package_single()
+        elif self.picking_type_id.automatic_package_creation_mode == "packaging":
+            self._auto_create_delivery_package_per_smallest_packaging()
+
+    def _auto_create_delivery_package_per_smallest_packaging(self) -> None:
+        """
+        Put each done smallest product packaging in a package
+        """
+        for picking in self:
+            for move_line in picking.move_line_ids:
+                if not move_line.qty_done:
+                    continue
+                if move_line.result_package_id:
+                    continue
+                packagings = move_line.product_id.packaging_ids
+                if not packagings:
+                    raise UserError(
+                        _(
+                            "Cannot create a package for the product %s as "
+                            "no product packaging is defined"
+                        )
+                        % move_line.product_id.display_name
+                    )
+                smallest_packaging = packagings.sorted("qty")[0]
+                precision_digits = self.env["decimal.precision"].precision_get(
+                    "Product Unit of Measure"
+                )
+                qty_done = move_line.qty_done
+                qty = float_round(
+                    qty_done / smallest_packaging.qty,
+                    precision_digits=precision_digits,
+                    rounding_method="HALF-UP",
+                )
+                if not qty.is_integer():
+                    raise UserError(
+                        _(
+                            "The done quantity of the product %s is not "
+                            "a multiple of product packaging"
+                        )
+                        % move_line.product_id.display_name
+                    )
+                for _i in range(int(qty)):
+                    move_line.qty_done = smallest_packaging.qty
+                    move_line.picking_id._put_in_pack(move_line)
+
+    def _auto_create_delivery_package_single(self) -> None:
+        """
+        For every move that don't have a package, set a new one.
+        """
+        for picking in self:
+            move_lines = picking.move_line_ids.filtered(
+                lambda s: not s.result_package_id
+            )
+            if move_lines:
+                picking._put_in_pack(move_lines)
```

Comment:
This would not do the same as it was initially. 
This would create one package for all of the move_lines. 
Initially it created a package for each move_lines qty_done
As example if a move_line's qty_done is 2 then it created tow packages, if it was 3 than 3 packages. 

```suggestion
    def _auto_create_delivery_package_single(self) -> None:
        """
        For every move that don't have a package, set a new one.
        """
        for picking in self:
            for move_line in picking.move_line_ids:
                if not self._is_auto_create_delivery_possible_for_move_line(move_line):
                    continue
                for qty in [1] * move_line.qty_done:
                    move_line.picking_id._put_in_pack(move_line)
```

Reply:
The behavior you are explaining is what is done by `_auto_create_delivery_package_per_smallest_packaging`.
This `single` method is a replacement of https://github.com/OCA/delivery-carrier/blob/14.0/base_delivery_carrier_label/models/stock_picking.py#L53 and https://github.com/OCA/delivery-carrier/blob/16.0/delivery_automatic_package/models/stock_picking.py#L12

Reply:
yes you are right, thx